### PR TITLE
Add docs for DisableRuntimeMarshalling

### PR DESCRIPTION
**Pull request by jlaanstra**
## Summary

Add docs for new DisableRuntimeMarshalling property as proposed in https://github.com/dotnet/sdk/pull/34388

/cc @AaronRobinsonMSFT 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/project-sdk/msbuild-props.md](https://github.com/dotnet/docs/blob/80e493c16c0f83d1d3487b092fdf59edc60ba558/docs/core/project-sdk/msbuild-props.md) | [MSBuild reference for .NET SDK projects](https://review.learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props?branch=pr-en-us-36497) |

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
@dotnet/interop-contrib Any concerns here? (specifically @elinor-fung)

File: docs/core/project-sdk/msbuild-props.md [hunk 1]
```diff
@@ -621,6 +623,23 @@ To speed up the build time, builds that are implicitly triggered by Visual Studi
...
 
+### DisableRuntimeMarshalling
+
+TODO
+
+### GenerateDisableRuntimeMarshallingAttribute
+
+The `GenerateDisableRuntimeMarshallingAttribute` property is closely related to the [DisableRuntimeMarshalling](#disableruntimemarshalling) property. 
+
+TODO
```
Comment (AaronRobinsonMSFT) [inline docs/core/project-sdk/msbuild-props.md:634]:
```suggestion
```

File: docs/core/project-sdk/msbuild-props.md [hunk 2]
```diff
@@ -621,6 +623,23 @@ To speed up the build time, builds that are implicitly triggered by Visual Studi
 </PropertyGroup>
 ```
 
+### DisableRuntimeMarshalling
+
+TODO
```
Comment (AaronRobinsonMSFT) [inline docs/core/project-sdk/msbuild-props.md:628]:
```suggestion
Build-in runtime marshalling can be disabled by applying the [`System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute`](xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute) attribute to an assembly. Guidance for disabling this feature can be found [here](../../standard/native-interop/disabled-marshalling.md).
```

File: docs/core/project-sdk/msbuild-props.md [hunk 3]
```diff
@@ -621,6 +622,17 @@ To speed up the build time, builds that are implicitly triggered by Visual Studi
 </PropertyGroup>
 ```
 
+### DisableRuntimeMarshalling
+
+Build-in runtime marshalling can be disabled by applying the [`System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute`](xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute) attribute to an assembly. Guidance for disabling this feature can be found [here](../../standard/native-interop/disabled-marshalling.md).
+
+
```
Comment (AaronRobinsonMSFT) [inline docs/core/project-sdk/msbuild-props.md:629]:
```suggestion
```

**Comment by AaronRobinsonMSFT** [comment]
@jkoritzinsky and @elinor-fung  Thoughts?

File: docs/core/project-sdk/msbuild-props.md [hunk 4]
```diff
@@ -621,6 +622,16 @@ To speed up the build time, builds that are implicitly triggered by Visual Studi
 </PropertyGroup>
 ```
 
+### DisableRuntimeMarshalling
+
+Build-in runtime marshalling can be disabled by applying the [`System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute`](xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute) attribute to an assembly. Guidance for disabling this feature can be found [here](../../standard/native-interop/disabled-marshalling.md).
```
Comment (lambdageek) [inline docs/core/project-sdk/msbuild-props.md:627]: nit
```suggestion
Built-in runtime marshalling can be disabled by applying the [`System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute`](xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute) attribute to an assembly. Guidance for disabling this feature can be found [here](../../standard/native-interop/disabled-marshalling.md).
```

**Review by lambdageek — COMMENTED** [review]
Is it worth mentioning why and when you might want to apply this attribute?  Something like "If your assembly has been converted to use the `LibraryImportAttribute` source generator instead of the runtime's builtin `DllImport` support then adding `DisableRuntimeMarshaling` will inform the runtime that built-in marshaling support is not needed. On some platforms, such as WebAssembly or iOS, if every assembly in a self-contained application does not require built-in marshaling, the runtime support may be removed at build-time and the app may accrue a modest size savings".

/cc @jandupej

**Comment by AaronRobinsonMSFT** [comment]
> Is it worth mentioning why and when you might want to apply this attribute?

That should be captured in https://learn.microsoft.com/dotnet/standard/native-interop/disabled-marshalling. This is merely documenting the existance of this well-known property.

**Review by jkoritzinsky — APPROVED** [review]

File: docs/core/project-sdk/msbuild-props.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (AaronRobinsonMSFT) [reply docs/core/project-sdk/msbuild-props.md]: I like this more than what I wrong. Will replace with yours.

File: docs/core/project-sdk/msbuild-props.md [hunk 5]
```diff
@@ -621,6 +622,16 @@ To speed up the build time, builds that are implicitly triggered by Visual Studi
 </PropertyGroup>
 ```
 
+### DisableRuntimeMarshalling
+
+Built-in runtime marshalling can be disabled by applying the [`System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute`](xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute) attribute to an assembly. Guidance for disabling this feature can be found [here](../../standard/native-interop/disabled-marshalling.md).
```
Comment (AaronRobinsonMSFT) [inline docs/core/project-sdk/msbuild-props.md:627]:
```suggestion
The `DisableRuntimeMarshalling` property enables you to specify that you would like to disable runtime marshalling support for your project. If this property is set to `true`, then the <xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute> will be added to the assembly and any P/Invokes or delegate-based interop will follow the rules for [disabled runtime marshalling](../../standard/native-interop/disabled-marshalling).
```

File: docs/core/project-sdk/msbuild-props.md [hunk 6]
```diff
@@ -621,6 +622,16 @@ To speed up the build time, builds that are implicitly triggered by Visual Studi
 </PropertyGroup>
 ```
 
+### DisableRuntimeMarshalling
+
+The `DisableRuntimeMarshalling` property enables you to specify that you would like to disable runtime marshalling support for your project. If this property is set to `true`, then the <xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute> will be added to the assembly and any P/Invokes or delegate-based interop will follow the rules for [disabled runtime marshalling](../../standard/native-interop/disabled-marshalling).
```
Comment (AaronRobinsonMSFT) [inline docs/core/project-sdk/msbuild-props.md:627]:
```suggestion
The `DisableRuntimeMarshalling` property enables you to specify that you would like to disable runtime marshalling support for your project. If this property is set to `true`, then the <xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute> will be added to the assembly and any P/Invokes or delegate-based interop will follow the rules for [disabled runtime marshalling](../../standard/native-interop/disabled-marshalling.md).
```

**Comment by AaronRobinsonMSFT** [comment]
@gewarren I think we are good from the tech side. Any feedback on phrasing?

File: docs/core/project-sdk/msbuild-props.md [hunk 7]
```diff
@@ -621,6 +622,16 @@ To speed up the build time, builds that are implicitly triggered by Visual Studi
 </PropertyGroup>
 ```
 
+### DisableRuntimeMarshalling
+
+The `DisableRuntimeMarshalling` property enables you to specify that you would like to disable runtime marshalling support for your project. If this property is set to `true`, then the <xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute> will be added to the assembly and any P/Invokes or delegate-based interop will follow the rules for [disabled runtime marshalling](../../standard/native-interop/disabled-marshalling.md).
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:627]:
```suggestion
The `DisableRuntimeMarshalling` property enables you to specify that you would like to disable runtime marshalling support for your project. If this property is set to `true`, then the <xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute> is added to the assembly and any P/Invokes or delegate-based interop will follow the rules for [disabled runtime marshalling](../../standard/native-interop/disabled-marshalling.md).
```

**Review by gewarren — APPROVED** [review]
LGTM with one nit.

**Pull request merged**